bluepysnap/edges.py

```diff
@@ -480,7 +482,37 @@ def _get(self, selection, properties=None):
 
         return result
 
-    def ids(self, group=None, limit=None, sample=None):
+    def _edge_ids_by_filter(self, queries, raise_missing_prop):
+        """Return edge IDs if their properties match the `queries` dict.
+
+        `props` values could be:
+            pairs (range match for floating dtype fields)
+            scalar or iterables (exact or "one of" match for other fields)
+
+        You can use the special operators '$or' and '$and' also to combine different queries
+        together.
+
+        Examples:
+            >>> self._edge_ids_by_filter({ Edge.POST_SECTION_ID: (0, 1),
+            >>>                            Edge.AXONAL_DELAY: (.5, 2.) })
+            >>> self._edge_ids_by_filter({'$or': [{ Edge.PRE_X_CENTER: [2, 3]},
+            >>>                              { Edge.POST_SECTION_POS: (0, 1),
+            >>>                              Edge.SYN_WEIGHT: (0.,1.4) }]})
+
+        """
+        properties = query.get_properties(queries)
+        unknown_props = properties - self.property_names
+        if raise_missing_prop and unknown_props:
+            raise BluepySnapError(f"Unknown edge properties: {unknown_props}")
+        res = []
+        ids = self.ids(None)
+        chunk_size = int(1e8)
+        for chunk in np.array_split(ids, 1 + len(ids) // chunk_size):
+            data = self.get(chunk, properties - unknown_props)
+            res.extend(chunk[query.resolve_ids(data, self.name, queries)])
+        return np.array(res, dtype=IDS_DTYPE)
```

Comment:
do you think we could have better perfs using smaller but more chunks and using multiprocessing (not dask just multiprocess)?

Reply:
I don't know. The problem here is the layer of libsonata between us and HDF5. If we go directly via `h5py` then such optimization would work. Otherwise, I am not sure.

Reply:
Hum good points ... Ok we can keep it like this.

Reply:
It is not necessary to merge it. We can abandon. I've implemented rather for history so we can have this discussion. If you still have some doubts, lets postpone or even abandon.

Reply:
No I think this is good to have this feature and we can improve it once it is here already. 

My only concern is to crash people pc when they are doing something they don t intend to. So the best would be to have something that can evaluate the size of the returned property dataframe in the function and compare it to the size of the user memory. So we can throw and clean memory before the ram + swap are full.